source/common/buffer/buffer_impl.cc

```diff
@@ -12,99 +13,317 @@
 namespace Envoy {
 namespace Buffer {
 
-// RawSlice is the same structure as evbuffer_iovec. This was put into place to avoid leaking
-// libevent into most code since we will likely replace evbuffer with our own implementation at
-// some point. However, we can avoid a bunch of copies since the structure is the same.
-static_assert(sizeof(RawSlice) == sizeof(evbuffer_iovec), "RawSlice != evbuffer_iovec");
-static_assert(offsetof(RawSlice, mem_) == offsetof(evbuffer_iovec, iov_base),
-              "RawSlice != evbuffer_iovec");
-static_assert(offsetof(RawSlice, len_) == offsetof(evbuffer_iovec, iov_len),
-              "RawSlice != evbuffer_iovec");
+static uint64_t SliceSize(uint64_t data_size) {
```

Comment:
If you make this a static method of OwnedBufferSlice then you can unit-test it. Also it should be sliceSize() whether a static method or static function.

Reply:
ping